Userland/Libraries/LibGUI/Dialog.cpp

```diff
@@ -25,16 +27,37 @@ int Dialog::exec()
 {
     VERIFY(!m_event_loop);
     m_event_loop = make<Core::EventLoop>();
-    if (parent() && is<Window>(parent())) {
-        auto& parent_window = *static_cast<Window*>(parent());
-        if (parent_window.is_visible()) {
-            center_within(parent_window);
+
+    auto desktop_rect = Desktop::the().rect();
+    auto window_rect = rect();
+
+    if (m_screen_position == CenterWithinParent) {
+        if (parent() && is<Window>(parent())) {
+            auto& parent_window = *static_cast<Window*>(parent());
+            if (parent_window.is_visible())
+                window_rect.center_within(parent_window.rect());
+            else
+                window_rect.center_within(desktop_rect);
         } else {
-            center_on_screen();
+            window_rect.center_within(desktop_rect);
         }
     } else {
-        center_on_screen();
+        if (m_screen_position & VerticalTop)
+            window_rect.set_y(32);
+        else if (m_screen_position & VerticalMiddle)
+            window_rect.center_vertically_within(desktop_rect);
+        else if (m_screen_position & VerticalBottom)
+            window_rect.set_y(desktop_rect.height() - Desktop::the().taskbar_height() - height() - 12);
+
+        if (m_screen_position & HorizontalLeft)
+            window_rect.set_x(12);
+        else if (m_screen_position & HorizontalMiddle)
+            window_rect.center_horizontally_within(desktop_rect);
+        else if (m_screen_position & HorizontalRight)
+            window_rect.set_x(desktop_rect.width() - width() - 12);
```

Comment:
This way of doing it makes for weird results when the user passes in just VerticalTop for example

Reply:
hm, what do you think we should default to for else case?

Reply:
Not really sure what you're meaning by default to for else case? My thought was using an enum or enum class with named positions (TopLeft, TopMiddle ..) and then checking it with a simple switch-case, this seem overly complicated.

Reply:
I did that, also added {top,bottom}_align + {left,right}_align to reduce duplicated code which otherwise couldn't be easily avoided with handling for all 9 cases.